doc/proposal/expected/d0323r3.md

```diff
@@ -163,82 +165,266 @@ Using `expected<double, errc>`, the code becomes:
 ```c++
 expected<double, errc> safe_divide(double i, double j)
 {
-	if (j==0) return unexpected(arithmetic_errc::divide_by_zero); // (1)
-	else return i / j; // (2)
+    if (j == 0) return unexpected(arithmetic_errc::divide_by_zero); // (1)
+    else return i / j; // (2)
 }
 ```
 
-(1) The implicit conversion from `unexpected<E>` to `expected<T,E>` and (2) from `T` to `expected<T,E>` prevents using too much boilerplate code. The advantages are that we have a clean way to fail without using the exception machinery, and we can give precise information about why it failed as well. The liability is that this function is going to be tedious to use. For instance, the exception-based 
+Note (1) the implicit conversion from `unexpected<E>` to `expected<T, E>` and
+(2) from `T` to `expected<T, E>` avoid boilerplate code. We have a clean way to
+fail without using the exception machinery, and we can give precise information
+about why it failed as well. The liability is that this function is going to be
+tedious to use. For instance, the exception-based function `i + j / k` is:
 
 ```c++
-function i + j/k is:
 double f1(double i, double j, double k)
 {
-	return i + safe_divide(j,k);
+    return i + safe_divide(j,k);
 }
+
 ```
 but becomes using `expected<double, errc>`:
 
 ```c++
 expected<double, errc> f1(double i, double j, double k)
 {
-	auto q = safe_divide(j, k)
-	if (q) return i + *q;
-	else return q;
+    auto q = safe_divide(j, k)
+    if (q) return i + *q;
+    else return q;
 }
 ```
-We can use `expected<T, E>` to represent different error conditions. For instance, with integer division, we might want to fail if the two numbers are not evenly divisible as well as checking for division by zero. We can overload our `safe_divide` function accordingly:
+
+We can use `expected<T, E>` to represent different error conditions. For
+instance, with integer division, we might want to fail if there's overflow, or
+if the two numbers are not evenly divisible, as well as checking for division by
+zero. We can overload our `safe_divide` function accordingly:
 
 ```c++
 expected<int, errc> safe_divide(int i, int j)
 {
-	if (j == 0) return unexpected(arithmetic_errc::divide_by_zero);
-	if (i%j != 0) return unexpected(arithmetic_errc::not_integer_division);
-	else return i / j;
+    if (j == 0) return unexpected(arithmetic_errc::divide_by_zero);
+    if (i == INT_MIN && j == -1) return unexpected(arithmetic_errc::integer_divide_overflows);
+    if (i % j != 0) return unexpected(arithmetic_errc::not_integer_division);
+    else return i / j;
+}
+```
+
+
+## URL parsing
+
+The following is how WebKit-based browsers [parse URLs] and use `std::expected`
+to denote failure:
+
+```c++
+template<typename CharacterType>
+std::expected<uint32_t, URLParser::IPv4PieceParsingError>
+URLParser::parseIPv4Piece(CodePointIterator<CharacterType>& iterator, bool& didSeeSyntaxViolation)
+{
```

Comment:
I'm worried that other examples will be dismissed as "toys". This is production code, deployed to a few million devices, with performance constraints. It's significantly harder to complain that the usecase isn't compelling with that type of code :)